compiler/src/dmd/dsymbolsem.d

```diff
@@ -6918,7 +6918,7 @@ private extern(C++) class SearchVisitor : Visitor
     {
         //printf("ArrayScopeSymbol::search('%s', flags = %d)\n", ident.toChars(), flags);
         if (ident != Id.dollar)
-            return setResult(null);
+            return visit(cast(ScopeDsymbol)ass);
```

Comment:
Not sure if this is controversial: it allows temporary symbols to be declared inside an "ArrayScope" that usually only contains the `$` symbol. This simplifies lowerings of expressions that are used as array indices.